modules/ebs_snapshot_explorer/main.py

```diff
@@ -0,0 +1,303 @@
+    while True:
+        response = input(prompt)
+        if response.lower() == 'y':
+            return True
+        elif response.lower() == 'n':
+            return False
+
+def load_volumes(client, print, input, instance_id, volume_ids):
+    """Loads volumes on an instance.
+
+    Args:
+        client (boto3.client): client to interact with AWS
+        print (func): Overwritten built-in print function
+        input (func): Overwritten built-in input function
+        instance_id (str): instance_id to attach volumes to
+        volume_ids (list): list of volume_ids to attach to the instance.
+    Returns:
+        bool: True if all volumes were successfully attached.
+    """
+
+    # load volume set
+    set_index = 0
+    set_count = 10
+
+    while set_index < len(volume_ids):
+        current_volume_set = volume_ids[set_index:set_index + set_count]
+        waiter = client.get_waiter('volume_available')
+        waiter.wait(VolumeIds=current_volume_set)
+        attached = modify_volume_set(
+            client, print, 'attach_volume', instance_id, current_volume_set)
+        if not attached:
+            print(' Volume attachment failed')
+            print(' Exiting...')
+            return False
+        running = input_helper(input)
+        detached = modify_volume_set(
+            client, print, 'detach_volume', instance_id, current_volume_set)
+        if not detached:
+            print(' Volume detachment failed')
+            print(' Exiting...')
+            return False
+        waiter.wait(VolumeIds=current_volume_set)
+        set_index += set_count
+        if not running:
+            break
+    cleanup(client)
+    return True
+
+
+def modify_volume_set(client, print, func, instance_id, volume_id_set):
+    """Helper function to load volumes on an instance to not overload the
+    instance.
+
+    Args:
+        client (boto3.client): client to interact with AWS
+        print (func): Overwritten built-in print function
+        func (str): Function sname to modify_volume_set
+        instance_id (str): instance_id to attach volumes to
+        volume_ids (list): list of volume_ids to (de)attach to the instance.
+    Returns:
+        bool: True if the volumes were successfully attached.
+    """
+    for volume_id in volume_id_set:
+        try:
+            kwargs = {
+                'InstanceId':instance_id,
+                'VolumeId':volume_id
+            }
+            if func == 'attach_volume':
+                kwargs['Device'] = get_valid_device(client, instance_id)
+            caller = getattr(client, func)
+            caller(**kwargs)
+        except ClientError as error:
+            code = error.response['Error']['Code']
+            if  code == 'UnauthorizedOperation':
+                print('  FAILURE MISSING AWS PERMISSIONS')
+            else:
+                print(error)
+            return False
+    return True
+
+def get_valid_device(client, instance):
```

Comment:
Just for clarity, it would probably make more sense to keep the name of this variable the same across different functions (instance_id instead of instance)